library/prelude/path.red

```diff
@@ -34,6 +34,20 @@ def funext
   λ i x →
   p _ i
 
+def happly 
+  (A : type) 
+  (B : A → type) 
+  (f g : (x : A) → B x) 
+  (p : path ((x : A) → B x) f g) 
+  : (x : A) → path (B x) (f x) (g x) = 
+  λ x i → p i x  
+
+def apd 
+  (A : type) (P : A → type) 
+  (f : (x : A) → P x) (x y : A) (p : path A x y) 
+  : path (P y) (coe 0 1 (f x) in λ i → P (p i)) (f y) = 
+  λ i → coe i 1 (f (p i)) in λ j → P (p j) 
```

Comment:
If we inline the use of `apd`, we can delete this definition.

Reply:
Sure, just thought to include it since it features in the book regularly later.